auto-dark.el

```diff
@@ -192,11 +202,26 @@ Remove theme change callback registered with D-Bus."
 
 (defun auto-dark--use-dbus ()
   "Determine whether we should use the dbus-* functions."
-  (and (fboundp 'dbus-register-signal)
-       (fboundp 'dbus-list-activatable-names)
-       (member "org.freedesktop.portal.Desktop"
-               (dbus-list-activatable-names :session))
-       t)) ;; t only here to make this a boolean
+  (eq auto-dark-detection-method 'dbus))
+
+(defun auto-dark--detect-detection-method ()
+  "Determine which theme detection method auto-dark should use."
+  (cond
+   ((and (eq system-type 'darwin)
+         (fboundp 'ns-do-applescript))
+    'applescript)
+   ((and (eq system-type 'darwin)
+         auto-dark-allow-osascript)
+    'osascript)
+   ((and (eq system-type 'gnu/linux)
+         (member 'dbus features)
```

Comment:
This checks if the `dbus` library was loaded.  We could swap this for `(require dbus nil t)` (and optionally remove that from the top of the file) and the functionality would be exactly the same.